### PR TITLE
Upgrade doc

### PR DESCRIPTION
**Pull request by joekelley**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1157

### CODE REVIEW

File: docs/content/operations/upgrading.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+that these steps will provide a comprehensive, zero-downtime upgrade experience. To reiterate:
+
+{{< notice warning >}}
+This upgrade process is **not guaranteed** to provide a comprehensive, zero-downtime upgrade experience.
+
+While Gloo Mesh is under initial development and in a pre-1.0.0 state, the recommended way to access new functionality
+is via a fresh install of Gloo Mesh.
+{{< /notice >}}
+
+1. Following the steps outlined by the [Getting Started Guide]({{% versioned_link_path fromRoot="/getting_started/" %}}),
```
Comment (jenshu) [inline docs/content/operations/upgrading.md:26]: the numbering seems off

File: docs/content/operations/upgrading.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+download the version of `meshctl` corresponding to the version of Gloo Mesh to which you would like to upgrade.
+
+1. Scale down both the `discovery` and `networking` deployments on the Gloo Mesh management cluster to zero replicas.
```
Comment (jenshu) [inline docs/content/operations/upgrading.md:29]: for this and the other scaling steps, should we write out the commands explicitly?

File: docs/content/operations/upgrading.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+This will prevent each component from processing resources structured in a way not expected by the new versions of each
+component.
+
+1. Delete all resources in the `discovery.gloo.mesh.gloo.solo.io` API group such as `meshes`, `traffictargets`, and `workloads`.
+These resources will be recreated when `discovery` is scaled back up later in the upgrade process. Deleting these
+resources ensures that the latest discovery implementation will recreate them with the structure it expects. 
+
+1. Manually update the Gloo Mesh management plane CRDs in the management cluster. These include both `discovery` and
```
Comment (jenshu) [inline docs/content/operations/upgrading.md:37]: can we list out the commands and maybe link to the CRDs they need to apply?

File: docs/content/operations/upgrading.md (hunk 1 above)
Comment (joekelley) [reply docs/content/operations/upgrading.md:26]: Markdown will render this as 1, 2, ... n -- makes it easier to re-order and iterate on the list items https://www.markdownguide.org/basic-syntax/#ordered-lists

Comment (jenshu) [reply docs/content/operations/upgrading.md:26]: 🤯

File: docs/content/operations/upgrading.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+# Add your desired version here
+UPGRADE_VERSION=v0.11.3
+
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/discovery.mesh.gloo.solo.io_v1alpha2_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/multicluster.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/networking.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/networking.mesh.gloo.solo.io_v1alpha2_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/rbac.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/settings.mesh.gloo.solo.io_v1alpha2_crds.yaml
+```
```
Comment (jenshu) [inline docs/content/operations/upgrading.md:75]: should this whole part go under step 4?

File: docs/content/operations/upgrading.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+```shell
+# Add your desired version here
+UPGRADE_VERSION=v0.11.3
+
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/discovery.mesh.gloo.solo.io_v1alpha2_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/multicluster.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/networking.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/networking.mesh.gloo.solo.io_v1alpha2_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/rbac.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds/settings.mesh.gloo.solo.io_v1alpha2_crds.yaml
```
Comment (jenshu) [inline docs/content/operations/upgrading.md:74]: could they apply everything in the directory as follows?
```
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh/crds
```

Comment (joekelley) [reply docs/content/operations/upgrading.md:74]: unfortunately i don't think so, https://raw.githubusercontent.com/solo-io/gloo-mesh/main/install/helm/gloo-mesh/crds 404s for example

File: docs/content/operations/upgrading.md (hunk 4 above)
Comment (joekelley) [reply docs/content/operations/upgrading.md:67]: yes good catch!

**Review by jenshu — APPROVED** [review]

**Pull request merged**